controllers/operandrequest/operandrequest_controller_test.go

```diff
@@ -160,27 +180,92 @@ var _ = Describe("OperandRegistry controller", func() {
+			Eventually(func() bool {
+				requestInstance2 := &operatorv1alpha1.OperandRequest{}
+				Expect(k8sClient.Get(ctx, requestKey2, requestInstance2)).Should(Succeed())
+				requestInstance2.Spec.Requests[0].Operands = requestInstance2.Spec.Requests[0].Operands[1:]
+				Expect(k8sClient.Update(ctx, requestInstance2)).Should(Succeed())
+				etcdCluster := &v1beta2.EtcdCluster{}
+				err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: "example", Namespace: operatorNamespaceName}, etcdCluster)
+				return err != nil && errors.IsNotFound(err)
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
 
-			By("Checking operators have been deleted")
+			By("Checking the etcd operator has been deleted")
 			Eventually(func() bool {
 				etcdCSV := &olmv1alpha1.ClusterServiceVersion{}
 				err := k8sClient.Get(ctx, types.NamespacedName{Name: "etcd-csv.v0.0.1", Namespace: operatorNamespaceName}, etcdCSV)
 				return err != nil && errors.IsNotFound(err)
 			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
 
+			By("Deleting the first OperandRequest")
+			Expect(k8sClient.Delete(ctx, request1)).Should(Succeed())
+
+			By("Checking jenkins operator has not been deleted")
+			Eventually(func() error {
+				jenkinsCSV := &olmv1alpha1.ClusterServiceVersion{}
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "jenkins-csv.v0.0.1", Namespace: operatorNamespaceName}, jenkinsCSV)
+				return err
+			}, testutil.Timeout, testutil.Interval).Should(Succeed())
+
+			By("Checking jenkins operator has not been deleted")
+			Eventually(func() bool {
+				jenkinsCSV := &olmv1alpha1.ClusterServiceVersion{}
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "jenkins-csv.v0.0.1", Namespace: operatorNamespaceName}, jenkinsCSV)
+				return err == nil
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
+
+			By("Checking jenkins operator has not been deleted")
+			Eventually(func() bool {
+				jenkinsCSV := &olmv1alpha1.ClusterServiceVersion{}
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "jenkins-csv.v0.0.1", Namespace: operatorNamespaceName}, jenkinsCSV)
+				return err != nil && errors.IsNotFound(err)
+			}, testutil.Timeout, testutil.Interval).Should(BeFalse())
+
+			By("Checking jenkins operator has been deleted")
+			Eventually(func() bool {
+				jenkinsCSV := &olmv1alpha1.ClusterServiceVersion{}
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "jenkins-csv.v0.0.1", Namespace: operatorNamespaceName}, jenkinsCSV)
+				return err == nil
+			}, testutil.Timeout, testutil.Interval).Should(BeFalse())
+
+			By("Checking jenkins operator has been deleted")
+			Eventually(func() bool {
+				jenkinsCSV := &olmv1alpha1.ClusterServiceVersion{}
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "jenkins-csv.v0.0.1", Namespace: operatorNamespaceName}, jenkinsCSV)
+				return err != nil && errors.IsNotFound(err)
+			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
```

Comment:
I used first three ways to check that `jenkins` operator is not deleted, it all pass the test.
and the last two are checking that  `jenkins` operator is deleted, it all pass the test as well.

So the only explanation I could image is that in the first the CSV exists but after waiting for a few, it is deleted.

Here is the passed test result from the last commit
http://prow.purple-chesterfield.com/log?job=test-operand-deployment-lifecycle-manager-amd64&id=1374220756690407425